src/AutoRest.CSharp/Mgmt/Decorator/TenantDetection.cs

```diff
@@ -23,6 +24,16 @@ public static bool IsTenantResource(this OperationGroup operationGroup, MgmtConf
             return result;
         }
 
+        public static bool IsTenantParent(this OperationGroup operationGroup, BuildContext context)
```

Comment:
should we call this, is Ancestor tenant?  Since you potentially are getting grand parent or even great grand parent, etc

Reply:
Renamed to `IsAncestorTenant`.